src/authres.rs

```diff
@@ -246,14 +293,33 @@ async fn compute_dkim_results(
     let ids_config = context.get_config(Config::AuthservIdCandidates).await?;
     let ids = parse_authservid_candidates_config(&ids_config);
 
-    // Remove all foreign authentication results
+    let origin = get_authservid_condidates_origin(context).await?;
+
+    // Remove all foreign Authentication-Results
     authres.retain(|(authserv_id, _dkim_passed)| ids.contains(authserv_id.as_str()));
 
     if authres.is_empty() {
-        // If the authentication results are empty, then our provider doesn't add them
-        // and an attacker could just add their own Authentication-Results, making us
-        // think that DKIM passed. So, in this case, we can as well assume that DKIM passed.
-        dkim_passed = true;
+        match origin {
+            AuthservIdCandidatesOrigin::None | AuthservIdCandidatesOrigin::UnknownSender => {
+                // If the Authentication-results are empty, then our provider doesn't add them
+                // and an attacker could just add their own Authentication-Results, making us
+                // think that DKIM passed. So, in this case, we can as well assume that DKIM passed.
+                dkim_passed = true;
+            }
+
+            AuthservIdCandidatesOrigin::SameDomainSender => {
+                // We know that usually our provider adds Authentication-Results,
+                // and we are quite certain that we know the correct authserv-id.
+                //
+                // Some providers like buzon.uy, disroot.org, yandex.ru, mailo.com, and
+                // riseup.net only add Authentication-Results if DKIM/SPF passed.
+                // So, probably DKIM failed, and therefore no Authentication-Results
+                // were added. And since we probably know the correct authserv-id,
+                // we can acutally assume that an attacher couldn't just add their own
+                // Authentication-Results.
+                dkim_passed = false;
+            }
```

Comment:
I just noticed that we can't do this, because it will sometimes incorrectly mark emails with a red exclamation mark if self's provider is e.g. fastmail.com, yahoo.com, aol.com or e.email:

These providers rotate the authserv-id. So, if we would merge this PR as-is, then we'll remember _one_ of the authserv-ids, as soon as we get an email from the same domain, and cling to it. 

Then as soon as we get an email from another domain, and it's signed with another authserv-id, we'll notice that there is no authres from "our" authserv-id, and therefore set `dkim_passed=false`. Since `dkim_passed` was `true` previously when the authres was from "our" authserv-id, we'll show the red exclamation mark (and deny key changes at some point).